File: app/src/main/java/ninja/dudley/yamr/ui/activities/Reader.java
package ninja.dudley.yamr.ui.activities;

import android.app.Activity;
import android.app.FragmentTransaction;
import android.content.res.Configuration;
import android.net.Uri;
import android.os.Bundle;
import android.util.Log;
import android.widget.RelativeLayout;

import ninja.dudley.yamr.R;
import ninja.dudley.yamr.ui.fragments.PageViewer;
import ninja.dudley.yamr.ui.fragments.ProviderViewer;
import ninja.dudley.yamr.ui.fragments.SeriesViewer;
import ninja.dudley.yamr.ui.util.OrientationAware;

public class Reader extends Activity implements ProviderViewer.LoadSeries, SeriesViewer.LoadChapter, OrientationAware.I
{
    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_reader);
        if (savedInstanceState == null)
        {
            FragmentTransaction transaction = getFragmentManager().beginTransaction();
            ProviderViewer providerViewer = new ProviderViewer();
            transaction.replace(R.id.reader, providerViewer);
            transaction.commit();
        }
    }

    @Override
    public void loadSeries(Uri series)
    {
        FragmentTransaction transaction = getFragmentManager().beginTransaction();
        SeriesViewer seriesViewer = new SeriesViewer();
        Bundle args = new Bundle();
        args.putParcelable(SeriesViewer.ArgumentKey, series);
        seriesViewer.setArguments(args);
        transaction.replace(R.id.reader, seriesViewer);
        transaction.addToBackStack(null);
        transaction.commit();
    }

    @Override
    public void loadChapter(Uri chapter)
    {
        FragmentTransaction transaction = getFragmentManager().beginTransaction();
        PageViewer pageViewer = new PageViewer();
        Bundle args = new Bundle();
        args.putParcelable(PageViewer.ChapterArgumentKey, chapter);
        pageViewer.setArguments(args);
        transaction.replace(R.id.reader, pageViewer);
        transaction.addToBackStack(null);
        transaction.commit();
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig)
    {
        OrientationAware.handleOrientationAware(this, newConfig);
        super.onConfigurationChanged(newConfig);
    }

    @Override
    public void onPortrait(Configuration newConfig)
    {
        RelativeLayout layout = (RelativeLayout) findViewById(R.id.reader);
        layout.removeAllViews();
    }

    @Override
    public void onLandscape(Configuration newConfig)
    {
        RelativeLayout layout = (RelativeLayout) findViewById(R.id.reader);
        layout.removeAllViews();
    }

    @Override
    protected void onSaveInstanceState(Bundle outState)
    {
        super.onSaveInstanceState(outState);
    }
}

File: app/src/main/kotlin/yamr/ui/activities/Reader.kt
package ninja.dudley.yamr.ui.activities

import android.app.Activity
import android.content.res.Configuration
import android.net.Uri
import android.os.Bundle
import android.widget.RelativeLayout
import ninja.dudley.yamr.R
import ninja.dudley.yamr.ui.fragments.ChapterViewer
import ninja.dudley.yamr.ui.fragments.PageViewer
import ninja.dudley.yamr.ui.fragments.ProviderViewer
import ninja.dudley.yamr.ui.fragments.SeriesViewer
import ninja.dudley.yamr.ui.util.OrientationAware

public class Reader : Activity(), ProviderViewer.LoadSeries, SeriesViewer.LoadChapter, ChapterViewer.LoadPage,  OrientationAware.I
{
    override fun onCreate(savedInstanceState: Bundle?)
    {
        super<Activity>.onCreate(savedInstanceState)
        setContentView(R.layout.activity_reader)
        if (savedInstanceState == null)
        {
            val transaction = getFragmentManager().beginTransaction()
            val providerViewer = ProviderViewer()
            transaction.replace(R.id.reader, providerViewer)
            transaction.commit()
        }
    }

    override fun loadSeries(series: Uri)
    {
        val transaction = getFragmentManager().beginTransaction()
        val seriesViewer = SeriesViewer()
        val args = Bundle()
        args.putParcelable(SeriesViewer.ArgumentKey, series)
        seriesViewer.setArguments(args)
        transaction.replace(R.id.reader, seriesViewer)
        transaction.addToBackStack(null)
        transaction.commit()
    }

    override fun loadChapter(chapter: Uri)
    {
        val transaction = getFragmentManager().beginTransaction()
        val chapterViewer = ChapterViewer(chapter)
        transaction.replace(R.id.reader, chapterViewer)
        transaction.addToBackStack(null)
        transaction.commit()
    }

    override fun loadFirstPageOfChapter(chapter: Uri)
    {
        val transaction = getFragmentManager().beginTransaction()
        val pageViewer = PageViewer()
        val args = Bundle()
        args.putParcelable(PageViewer.ChapterArgumentKey, chapter)
        pageViewer.setArguments(args)
        transaction.replace(R.id.reader, pageViewer)
        transaction.addToBackStack(null)
        transaction.commit()
    }

    override fun loadPage(page: Uri)
    {
        val transaction = getFragmentManager().beginTransaction()
        val pageViewer = PageViewer()
        val args = Bundle()
        args.putParcelable(PageViewer.PageArgumentKey, page)
        pageViewer.setArguments(args)
        transaction.replace(R.id.reader, pageViewer)
        transaction.addToBackStack(null)
        transaction.commit()
    }

    override fun onConfigurationChanged(newConfig: Configuration)
    {
        OrientationAware.handleOrientationAware(this, newConfig)
        super<Activity>.onConfigurationChanged(newConfig)
    }

    override fun onPortrait(newConfig: Configuration)
    {
        val layout = findViewById(R.id.reader) as RelativeLayout
        layout.removeAllViews()
    }

    override fun onLandscape(newConfig: Configuration)
    {
        val layout = findViewById(R.id.reader) as RelativeLayout
        layout.removeAllViews()
    }

    override fun onSaveInstanceState(outState: Bundle)
    {
        super<Activity>.onSaveInstanceState(outState)
    }
}
